src/server/auth.ts

```diff
@@ -75,10 +84,34 @@ export const authOptions: NextAuthOptions = {
     },
   },
   providers: [
-    GoogleProvider({
-      clientId: env.GOOGLE_CLIENT_ID,
-      clientSecret: env.GOOGLE_CLIENT_SECRET,
-    }),
+    process.env.VERCEL_ENV === 'preview'
+      ? CredentialsProvider({
+          name: 'Credentials',
+          credentials: {
+            username: {
+              label: 'Username',
+              type: 'text',
+              placeholder: 'jsmith',
+            },
+            password: { label: 'Password', type: 'password' },
+          },
+          authorize: () => {
+            return {
+              id: '1',
+              name: 'J Smith',
+              email: 'jsmith@example.com',
+              image: 'https://picsum.photos/id/237/200/300',
+            } as PreviewUser;
+          },
+        })
+      : (GoogleProvider({
+          clientId: env.GOOGLE_CLIENT_ID,
+          clientSecret: env.GOOGLE_CLIENT_SECRET,
+        }),
+        DiscordProvider({
+          clientId: process.env.DISCORD_CLIENT_ID ?? '',
+          clientSecret: process.env.DISCORD_CLIENT_SECRET ?? '',
```

Comment:
Make sure to just use `env` since its the typesafe object

Reply:
For some reason they can be undefined. Should I just use '!'?